File: go-kata/list_test.go
package main

import (
	"kata-machine/interfaces"
	"testing"

	"github.com/stretchr/testify/assert"
)

func testList(t *testing.T, list interfaces.List[int]) {
	list.Append(5)
	list.Append(7)
	list.Append(9)

	assert.Equal(t, 9, *list.Get(2))
	assert.Equal(t, 7, *list.RemoveAt(1))
	assert.Equal(t, 2, list.Length())

	list.Append(11)
	assert.Equal(t, 9, *list.RemoveAt(1))
	assert.Nil(t, list.Remove(9))
	assert.Equal(t, 5, *list.RemoveAt(0))
	assert.Equal(t, 11, *list.RemoveAt(0))
	assert.Equal(t, 0, list.Length())

	list.Prepend(5)
	list.Prepend(7)
	list.Prepend(9)

	assert.Equal(t, 5, *list.Get(2))
	assert.Equal(t, 9, *list.Get(0))
	assert.Equal(t, 9, *list.Remove(9))
	assert.Equal(t, 2, list.Length())
	assert.Equal(t, 7, *list.Get(0))
}

File: java-kata/src/test/java/com/kata/ListTest.java
package com.kata;

import com.kata.interfaces.List;
import static org.junit.jupiter.api.Assertions.*;

public class ListTest {
    
    public static void testList(List<Integer> list) {
        list.append(5);
        list.append(7);
        list.append(9);

        assertEquals(9, list.get(2));
        assertEquals(7, list.removeAt(1));
        assertEquals(2, list.getLength());

        list.append(11);
        assertEquals(9, list.removeAt(1));
        assertNull(list.remove(9));
        assertEquals(5, list.removeAt(0));
        assertEquals(11, list.removeAt(0));
        assertEquals(0, list.getLength());

        list.prepend(5);
        list.prepend(7);
        list.prepend(9);

        assertEquals(5, list.get(2));
        assertEquals(9, list.get(0));
        assertEquals(9, list.remove(9));
        assertEquals(2, list.getLength());
        assertEquals(7, list.get(0));
    }
}